src/main/java/uk/gov/pay/api/model/TransactionResponse.java

```diff
@@ -0,0 +1,172 @@
+
+    private PaymentState state;
+
+    private String description;
+
+    private String reference;
+
+    private String email;
+
+    @JsonDeserialize(using = CustomSupportedLanguageDeserializer.class)
+    private SupportedLanguage language;
+
+    @JsonProperty(value = "delayed_capture")
+    private boolean delayedCapture;
+
+    @JsonProperty("corporate_card_surcharge")
+    private Long corporateCardSurcharge;
+
+    @JsonProperty("total_amount")
+    private Long totalAmount;
+
+    @JsonProperty("fee")
+    private Long fee;
+
+    @JsonProperty("net_amount")
+    private Long netAmount;
+
+    @JsonProperty(value = "created_date")
+    private String createdDate;
+
+    @JsonProperty(value = "gateway_transaction_id")
+    private String gatewayTransactionId;
+
+    @JsonDeserialize(using = ExternalMetadataDeserialiser.class)
+    private ExternalMetadata metadata;
+
+    public Optional<ExternalMetadata> getMetadata() {
+        return Optional.ofNullable(metadata);
+    }
+
+    public String getChargeId() {
+        return chargeId;
+    }
+
+    public Long getAmount() {
+        return amount;
+    }
+
+    public PaymentState getState() {
+        return state;
+    }
+
+    public String getReturnUrl() {
+        return returnUrl;
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    public String getReference() {
+        return reference;
+    }
+
+    public String getEmail() {
+        return email;
+    }
+
+    public SupportedLanguage getLanguage() {
+        return language;
+    }
+
+    public boolean getDelayedCapture() {
+        return delayedCapture;
+    }
+
+    public Long getCorporateCardSurcharge() {
+        return corporateCardSurcharge;
+    }
+
+    public Long getTotalAmount() {
+        return totalAmount;
+    }
+    
+    public Long getFee() {
+        return fee;
+    }
+
+    public Long getNetAmount() {
+        return netAmount;
+    }
+
+    public String getPaymentProvider() {
+        return paymentProvider;
+    }
+
+    /**
+     * card brand is no longer a top level charge property. It is now at `card_details.card_brand` attribute
+     * We still need to support `v1` clients with a top level card brand attribute to keep support their integrations.
+     *
+     * @return
+     */
+    @JsonProperty("card_brand")
+    public String getCardBrand() {
+        return cardDetails != null ? cardDetails.getCardBrand() : "";
+    }
+
+    public String getCreatedDate() {
+        return createdDate;
+    }
+
+    public List<PaymentConnectorResponseLink> getLinks() {
+        return links;
+    }
+
+    public RefundSummary getRefundSummary() {
+        return refundSummary;
+    }
+
+    public SettlementSummary getSettlementSummary() {
+        return settlementSummary;
+    }
+
+    public CardDetails getCardDetails() {
+        return cardDetails;
+    }
+
+    public String getGatewayTransactionId() {
+        return gatewayTransactionId;
+    }
+}
```

Comment:
My understanding is that the purpose of introducing TransactionResponse and Charge is to allow you to create an abstraction over the two different apis, so that you get a 
`ChargeFromResponse` from connector
and a
`TransactionResponse` from ledger
and both can be converted into a `Charge`
I'm not sure that we need to do this - it seems the structure of all three of these  classes is basically the same? I may have got the wrong end of the stick though, so let's discuss this.

Reply:
Your understanding is correct.

At the moment the structure is the same, but I would prefer no to tie ledger API to connector API (it seems to risky).
It's not always easy to say when it's better to have deliberate duplication of the code to make sure we don't mix things up.